database/database.go

```diff
@@ -1,127 +1,585 @@
+
+}
+
+// StoreUnverifiedQuote stores an unverified quote
+func StoreUnverifiedQuote() {
+
+}
+
+// DeleteUnverifiedQuote deletes an unverified quote
+func DeleteUnverifiedQuote() {
+
 }
 
-// GetQuotes is a function to get quotes from the database
-func GetQuotes() ([]Quote, error) {
+/* -------------------------------------------------------------------------- */
+/*                              PRIVATE FUNCTIONS                             */
+/* -------------------------------------------------------------------------- */
+
+func loadLocalDatabase() error {
+	var err error
+
+	log.Print("Creating localDatabase from PostgreSQL database...")
 
-	rows, err := database.Query("SELECT quotes.id, quotes.text, teachers.id, teachers.name, teachers.title, teachers.note FROM quotes INNER JOIN teachers ON quotes.teacherid = teachers.id")
-	defer rows.Close()
+	// initialize characterLookup table
+	setupCharacterLookup()
 
+	// Verify connection to PostgreSQL database
+	err = postgresDatabase.Ping()
 	if err != nil {
-		log.Print("Cannot get quotes: ", err)
-		return nil, err
+		return errors.New("At createLocalDatabase: " + err.Error())
 	}
 
-	var quotes []Quote
+	localDatabase.mux.LockWrite()
+	defer localDatabase.mux.UnlockWrite()
 
+	/* --------------------------------- QUOTES --------------------------------- */
+
+	// get all quotes from PostgreSQL database
+	rows, err := postgresDatabase.Query(`SELECT 
+		QuoteID,
+		TeacherID, 
+		Context,
+		Text,
+		Unixtime,
+		Upvotes FROM quotes`)
+
+	if err != nil {
+		return errors.New("At createLocalDatabase: " + err.Error())
+	}
+
+	// initialize wordsMap of localDatabase
+	localDatabase.wordsMap = make(map[string]wordsMapT)
+
+	// Iterrate over all quotes from PostgreSQL database
+	for rows.Next() {
+		// Get id and text of quote
+		var q QuoteT
+		rows.Scan(&q.QuoteID, &q.TeacherID, &q.Context, &q.Text, &q.Unixtime, &q.Upvotes)
+
+		// add to local database
+		// unsafe, because localDatabase is already locked for writing
+		err = unsafeAddQuoteToLocalDatabase(q)
+		if err != nil {
+			return errors.New("At createLocalDatabase: " + err.Error())
+		}
+	}
+
+	rows.Close()
+
+	/* -------------------------------- TEACHERS -------------------------------- */
+
+	// get all teachers from PostgreSQL database
+	rows, err = postgresDatabase.Query(`SELECT
+		TeacherID, 
+		Name, 
+		Title, 
+		Note FROM teachers`)
+
+	if err != nil {
+		return errors.New("At createLocalDatabase: " + err.Error())
+	}
+
+	// Iterate over all teachers from PostgreSQL database
 	for rows.Next() {
-		q := Quote{}
-		rows.Scan(&q.ID, &q.Text, &q.Teacher.ID, &q.Teacher.Name, &q.Teacher.Title, &q.Teacher.Note)
-		quotes = append(quotes, q)
+		// Get id and text of quote
+		var t TeacherT
+		rows.Scan(&t.TeacherID, &t.Name, &t.Title, &t.Note)
+
+		// add to local database
+		// unsafe, because localDatabase is already locked for writing
+		unsafeAddTeacherToLocalDatabase(t)
+	}
+
+	rows.Close()
+
+	log.Print("Done")
```

Comment:
clarify what "Done" means, e.g.
```suggestion
	log.Print("Finished loading Cache")
```